kratos/utilities/time_discretization.h

```diff
@@ -0,0 +1,128 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher
+//
+
+#if !defined( KRATOS_TIME_DISCRETIZATION_H_INCLUDED )
+#define  KRATOS_TIME_DISCRETIZATION_H_INCLUDED
+
+// System includes
+#include <array>
+
+// External includes
+
+// Project includes
+
+namespace Kratos {
+namespace TimeDiscretization {
+
+class KRATOS_API(KRATOS_CORE) BDF1
+{
+public:
+    std::array<double, 2> ComputeBDFCoefficients(double DeltaTime) const;
+};
+
+class KRATOS_API(KRATOS_CORE) BDF2
+{
+public:
+    std::array<double, 3> ComputeBDFCoefficients(double DeltaTime,
+                                                 double PreviousDeltaTime) const;
+};
+
+class KRATOS_API(KRATOS_CORE) BDF3
+{
+public:
+    std::array<double, 4> ComputeBDFCoefficients(double DeltaTime) const;
+};
+
+class KRATOS_API(KRATOS_CORE) BDF4
+{
+public:
+    std::array<double, 5> ComputeBDFCoefficients(double DeltaTime) const;
+};
+
+class KRATOS_API(KRATOS_CORE) BDF5
+{
+public:
+    std::array<double, 6> ComputeBDFCoefficients(double DeltaTime) const;
+};
+
+class KRATOS_API(KRATOS_CORE) BDF6
+{
+public:
+    std::array<double, 7> ComputeBDFCoefficients(double DeltaTime) const;
+};
+
+class Newmark
+{
+public:
+    // Attention, when changing the defaults do this also in the python-interface!
+    // pybind cannot detect them automatically!
+    Newmark(const double NewmarkBeta=0.25)
+        : mNewmarkBeta(NewmarkBeta) {}
+    double GetBeta()  const { return mNewmarkBeta; }
+    double GetGamma() const { return 0.5; }
+private:
+    double mNewmarkBeta;
+};
+
+class Bossak
+{
+public:
+    // Attention, when changing the defaults do this also in the python-interface!
+    // pybind cannot detect them automatically!
+    Bossak(const double AlphaM=-0.3,
+           const double NewmarkBeta=0.25)
+        : mAlphaM(AlphaM),
+          mNewmarkBeta(NewmarkBeta) {}
+    double GetAlphaM() const { return mAlphaM; }
+    double GetBeta()   const { return 0.5 + mAlphaM; }
+    double GetGamma()  const { return mNewmarkBeta * (1+mAlphaM) * (1+mAlphaM); }
+private:
+    double mAlphaM;
```

Comment:
Consider adding ctors `explicit Bossak(double AlphaM)` and `Bossak()` and putting in-class defaults. I think this way you can avoid the danger of somebody changing  a default here, but not in the pybind.
```suggestion
    double mAlphaM=-0.3;
```

Reply:
sure ...
Why did I not think of this ...